modules/apps/headless/headless-user-notification/headless-user-notification-test/src/testFunctional/macros/UserNotificationAPI.macro

```diff
@@ -0,0 +1,128 @@
+definition {
+
+	macro _curlNotificationTemplates {
+		var portalURL = JSONCompany.getPortalURL();
+
+		if (isSet(notificationTemplateId)) {
+			var api = "notification-templates/${notificationTemplateId}";
+		}
+		else {
+			var api = "notification-templates";
+		}
+
+		var curl = '''
+			${portalURL}/o/notification/v1.0/${api} \
+				-u test@liferay.com:test \
+				-H accept: application/json \
+				-H 'Content-Type: application/json' \
+		''';
+
+		return ${curl};
+	}
+
+	macro _curlUserNotifications {
+		var portalURL = JSONCompany.getPortalURL();
+
+		if (isSet(userNotificationId)) {
+			var api = "user-notifications/${userNotificationId}";
+
+			if (isSet(parameter)) {
+				var api = "user-notifications/${userNotificationId}/${parameter}";
+			}
+		}
+		else if (isSet(userAccountId)) {
+			var api = "user-accounts/${userAccountId}/user-notifications";
+		}
+		else {
+			var api = "my-user-notifications";
+		}
+
+		var curl = '''
+			${portalURL}/o/headless-user-notification/v1.0/${api} \
+				-H accept: application/json \
+				-H 'Content-Type: application/json' \
+				-u test@liferay.com:test \
+		''';
+
+		return ${curl};
+	}
+
+	macro createNotificationTemplate {
+		Variables.assertDefined(parameterList = "${name},${recipientType},${userScreenName},${en_US},${type}");
+
+		var curl = UserNotificationAPI._curlNotificationTemplates();
+		var body = '''
+			-d {
+				"editorType": "richText",
+				"externalReferenceCode": "${externalReferenceCode}",
+				"name": "${name}",
+				"recipientType": "${recipientType}",
+				"recipients": [
+					{
+						"userScreenName": "${userScreenName}"
+					}
+				],
+				"subject": {
+					"en_US": "${en_US}"
+				},
+				"type": "${type}"
+			}
+		''';
+
+		var curl = StringUtil.add(${curl}, ${body}, " ");
+
+		var curl = JSONCurlUtil.post(${curl});
+
+		var notificationTemplateId = JSONUtil.getWithJSONPath(${curl}, "$.id");
+
+		static var staticNotificationTemplateId = ${notificationTemplateId};
+
+		return ${staticNotificationTemplateId};
+	}
+
+	macro deleteNotificationTemplate {
+		var curl = UserNotificationAPI._curlNotificationTemplates(notificationTemplateId = ${staticNotificationTemplateId});
+
+		JSONCurlUtil.delete(${curl});
+	}
+
+	macro getUserAccountUserNotifications {
+		Variables.assertDefined(parameterList = ${userAccountId});
+
+		var curl = UserNotificationAPI._curlUserNotifications(userAccountId = ${userAccountId});
+
+		var response = JSONCurlUtil.get(${curl});
+
+		return ${response};
+	}
+
+	macro getUserNotifications {
+		if (isSet(userNotificationId)) {
```

Comment:
We already use if(isSet(userNotificationId)) in the _curlUserNotifications(), so I think we don't need to write it again since we don't define paramterList here, I mean remove the if/else here may get the same results we expected, could you try this and update if it works? Thanks
```
macro getUserNotifications {
			var curl = UserNotificationAPI._curlUserNotifications(userNotificationId = ${userNotificationId});

		var response = JSONCurlUtil.get(${curl});

		return ${response};
	}
```

Reply:
Update!